builtin/mainmenu/dlg_contentstore.lua

```diff
@@ -1110,7 +1169,23 @@ function store.handle_submit(this, fields)
 	return false
 end
 
-function create_store_dlg(type)
+function store.handle_events(event)
+	if event == "DialogShow" then
+		do_auto_install_pkg()
```

Comment:
Wasn't immediately clear to me why this DialogShow callback was needed

```suggestion
		-- If the store is already loaded, auto-install packages here
		do_auto_install_pkg()
```

Reply:
Is there a reason why you handle this case here rather than in the constructor of the dialog?

```lua
store.load()

-- Handle auto install if store is already loaded
do_auto_install_pkg()
```

Reply:
prepare_auto_install_pkg is only called in load, so if you load the menu once and then attempt to autoinstall it won't work. This will cause issues when using this feature in #13807

Could prepare_auto_install_pkg and do_auto_install_pkg be merged into a single instruction, where auto_install_pkg is only ever the key?

Reply:
> Is there a reason why you handle this case here rather than in the constructor of the dialog?

Yes, the dialog isn't visible yet when the constructor runs, and `do_auto_install_pkg` needs the dialog to be visible. Also, I think this is cleaner.